tests/pytests/integration/states/test_highstate_paths.py

```diff
@@ -0,0 +1,305 @@
+"""
+Tests for the highstate state relative and absolute paths
+"""
+import logging
+
+import pytest
+
+log = logging.getLogger(__name__)
+
+
+pytestmark = [
+    pytest.mark.slow_test,
+    pytest.mark.windows_whitelisted,
+]
+
+
+@pytest.fixture(scope="module")
+def pillar_tree(salt_master, salt_minion, salt_call_cli):
+    top_file = """
+    base:
+      '{}':
+        - basic
+    """.format(
+        salt_minion.id
+    )
+    basic_pillar_file = """
+    monty: python
+    companions:
+      three:
+        - liz
+        - jo
+        - sarah jane
+    """
+    top_tempfile = salt_master.pillar_tree.base.temp_file("top.sls", top_file)
+    basic_tempfile = salt_master.pillar_tree.base.temp_file(
+        "basic.sls", basic_pillar_file
+    )
+
+    try:
+        with top_tempfile, basic_tempfile:
+            ret = salt_call_cli.run("saltutil.refresh_pillar", wait=True)
+            assert ret.exitcode == 0
+            assert ret.json is True
+            yield
+    finally:
+        # Refresh pillar again to cleaup the temp pillar
+        ret = salt_call_cli.run("saltutil.refresh_pillar", wait=True)
+        assert ret.exitcode == 0
+        assert ret.json is True
+
+
+def test_rel_abs_path_ids(
+    salt_master,
+    salt_call_cli,
+    pillar_tree,
+    tmp_path,
+    salt_minion,
+):
+    """
+    This tests for any regressions for this issue:
+    https://github.com/saltstack/salt/issues/54179
+    """
+    test_tempdir = salt_master.state_tree.base.paths[0] / "tmp_dir"
+    test_tempdir.mkdir(parents=True, exist_ok=True)
+
```

Comment:
```suggestion
```
The directory should automatically be created. It didn't for you?

Reply:
https://github.com/saltstack/pytest-salt-factories/blob/master/src/saltfactories/utils/tempfiles.py#L147

Reply:
done